docs/zeebe/dynamic_scaling.md

```diff
@@ -0,0 +1,209 @@
+
+CoordinatorNode ->> P-1: Redistribution completed?
+P-1 ->> CoordinatorNode: No
+
+P-1 ->> P-NEW: Snapshot for bootstrap with SCALING_UP event
+P-NEW ->>- P-1: Partition NEW Bootstrapped
+P-1 ->> P-1: Start distributing pending commands
+
+loop for command in enqueued_distribution_commands
+  P-1 ->> P-NEW: send command 
+end
+
+P-1 ->> P-2: Partition i Bootstrapped
+CoordinatorNode ->> P-1: Redistribution completed?
+P-1 ->> CoordinatorNode: Yes
+CoordinatorNode ->>- CoordinatorNode : Update RoutingState.RequestHandling
+
+P-1 ->> P-ith: Distribution event at b_e + n
+P-1 ->> P-NEW: Distribution event at b_e + n
+```
+
+### Command Distribution Mechanism
+
+The redistribution phase uses [Zeebe's command distribution](./generalized_distribution.md) system to ensure reliable deployment of resources across partitions:
+
+1. **Command Enqueueing**: When scaling begins, deployment commands for new partitions are enqueued but not immediately sent
+2. **Bootstrap Completion Signal**: Each new partition signals completion of bootstrap to partition 1
+3. **Distributed Deployment**: When a partition is bootstrapped, partition 1 distributes queued deployment commands to the partitions (including existing ones)
+4. **Acknowledgment**: Receiving partitions acknowledge successful processing of deployment commands
+5. **Retry Logic**: Failed distributions are automatically retried with exponential backoff
+
+### Scaling steps
+
+1. A client sends a [ClusterConfigurationManagementRequest](../../zeebe/dynamic-config/src/main/java/io/camunda/zeebe/dynamic/config/api/ClusterConfigurationManagementApi.java) as a HTTP PATCH request at `/actuator/cluster/` on the management port.
+
+   If new brokers need to be added, it can be done in the same command, by specifying the ids of the brokers to add:
+   - with brokers:
+
+   ```bash
+   curl -X 'PATCH' \
+   'http://localhost:9600/actuator/cluster' \
+   -H 'accept: application/json' \
+   -H 'Content-Type: application/json' \
+   -d '{  
+   "brokers": { 
+   "add": [3,4,5]
+   },
+   "partitions": {
+   "count": 6, 
+   "replicationFactor": 3
+   }
+   }' | jq
+   ```
+
+   - without brokers
+
+   ```bash
+   curl -X 'PATCH' \
+   'http://localhost:9600/actuator/cluster' \
+   -H 'accept: application/json' \
+   -H 'Content-Type: application/json' \
+   -d '{  
+   "brokers": { 
+   "add": [3,4,5]
+   },
+   "partitions": {
+   "count": 6, 
+   "replicationFactor": 3
+   }
+   }' | 
+   ```
+2. The [ClusterChangeCoordinator](../../zeebe/dynamic-config/src/main/java/io/camunda/zeebe/dynamic/config/changes/ConfigurationChangeCoordinator.java) generates the `ClusterChangeOperation` needed to reach the requested result.
+   For a pure scaling operation, the generated `ClusterChangeOperation` will be:
+   - `StartPartitionScaleUp` to Partition 1
+   - `PartitionBootstrapOperation` to all new partitions
+   - `AwaitRedistributionCompletion` to the coordinator node 1
```

Comment:
:x: Coordinator node (broker  0) executes this operation by querying Partition 1.